xcm/xcm-executor/src/lib.rs

```diff
@@ -304,7 +323,7 @@ impl<Config: config::Config> XcmExecutor<Config> {
 
 	/// Execute any final operations after having executed the XCM message.
 	/// This includes refunding surplus weight, trapping extra holding funds, and returning any errors during execution.
-	pub fn post_execute(mut self, xcm_weight: Weight) -> Outcome {
+	pub fn post_execute(mut self, xcm_weight: Weight, message_hash: [u8; 32]) -> Outcome {
```

Comment:
`message_hash` should probably be stored in `self` rather than having the caller pass it in twice since it was already passed by the caller in `execute`/`execute_with_hash`.